lib/segment/src/index/field_index/field_index_base.rs

```diff
@@ -116,6 +117,7 @@ pub trait ValueIndexer<T> {
 #[allow(clippy::enum_variant_names)]
 pub enum FieldIndex {
     IntIndex(NumericIndex<IntPayloadType>),
+    DatetimeIndex(NumericIndex<IntPayloadType>),
```

Comment:
Is this enum variant really needed? I'm thinking maybe we can redirect to `IntIndex` for datetime payloads without this.

Reply:
It's used here to select the appropriate parser. (for `DatetimeIndex` we should parse strings, and for `IntIndex` we should get index).
https://github.com/qdrant/qdrant/blob/c5bc3aef6619bb95b5cd1f61ca20dbd2e345008a/lib/segment/src/index/field_index/field_index_base.rs#L270-L275

I see three ways to handle this:
1. Add a new enum variant `FieldIndex::DatetimeIndex`. (initial implementation of this PR)
2. Use numeric values and strings interchangeably. I think it's a too loose way.
3. Select parser using both `FieldIndex` and `PayloadFieldSchema`. I've updated the PR to pass `PayloadFieldSchema` for this. (commits `Introduce IndexMapItem` and `Drop FieldIndex::DatetimeIndex` )

Reply:
I liked solution #1 better. In this case it makes total sense for me to add the variant. Sorry for missing it :D